docs/syntax-jsx-support.md

```diff
@@ -0,0 +1,338 @@
+
+- `jsx = React.createElement`
+- `jsx = Roact.createElement`
+- `jsx = MyCustomElementFactory`
+
+This keeps the language neutral and allows different runtimes to adopt the syntax without coupling Luau to any one library.
+
+#### Intrinsics mapping for lowercase tags
+
+Because lowercase tags lower to string values (e.g. `<frame />` → `jsx("frame", ...)`), most runtimes will want a small **lookup table** that maps these “intrinsic” names to the runtime’s native element types (for Roblox, typically Instance class names such as `"Frame"`).
+
+For example:
+
+```lua
+local Intrinsics = {
+    frame = "Frame",
+    textlabel = "TextLabel",
+    uilistlayout = "UIListLayout",
+    -- ... (many more)
+}
+
+function jsx(tag, props, ...)
+    if type(tag) == "string" then
+        local className = Intrinsics[tag]
+        assert(className, ("Unknown intrinsic tag %q"):format(tag))
+        -- dispatch to your runtime using className
+    else
+        -- tag is a component (function/table/etc.) -> dispatch accordingly
+    end
+end
+```
+
+This approach is explicit and predictable, avoids ambiguous casing rules, and allows aliases (`div = "Frame"`) if desired.
+
+### Example: binding JSX to Fusion (community framework)
+
+Fusion is a popular community framework for Roblox that uses a declarative style for Instances and reactive state. Without changing the JSX proposal, a project can adopt JSX by binding `jsx` to a Fusion element factory.
+
+In practice, Fusion code commonly looks like `Fusion.New("Frame"){ ... }` and uses special keys like `Fusion.Children` for child lists. The exact adapter shape may vary by project, but a representative sketch is:
+
+```lua
+local Fusion = require(Packages.Fusion)
+local New = Fusion.New
+local Children = Fusion.Children
+
+local Intrinsics = {
+    textlabel = "TextLabel",
+    frame = "Frame",
+    -- ...
+}
+
+-- One possible adapter shape:
+--   jsx("frame", props, ...children) -> Fusion.New("Frame")(props)
+--   jsx(FooComponent, props, ...children) -> FooComponent(props, ...children) (project-defined)
+function jsx(tag, props, ...)
+    props = props or {}
+    props[Children] = { ... }
+
+    if type(tag) == "string" then
+        return New(Intrinsics[tag])(props)
+    else
+        -- component case is project-defined; this is just a sketch
+        return tag(props)
+    end
+end
+
+local State = Fusion.Value("Hello")
+
+local ui =
+    <textlabel
+        Text={State}
+        Size={UDim2.fromScale(1, 1)}
+    />
+```
+
+This example illustrates the key point: **JSX is a “call tree” surface syntax**, and Fusion (not Luau) defines the semantics.
+
+### Example: non-UI usage (scene graph / “fiber-like” tree)
+
+JSX can also be used for non-UI declarative graphs, e.g. building a 3D scene tree out of Roblox Instances (a “fiber-like” idea: a declarative tree that produces runtime objects).
```

Comment:
I don't really think we need to have an argument about the idea that XML syntax, whose entire raison d'etre is "authoring trees," can represent trees. Nobody is disputing that, the points of contention, as you can hopefully tell from the rest of the comments here, would be things like "what does embedded XML syntax in Luau mean?" and "does it even make sense for the language to concern itself with hybridizing itself with another language?" Those are the questions that this RFC needs to answer, and it does the former (though with details I think are insufficient to make the cut for the language), and seems to make no argument for the latter.

Reply:
Fair. Will make a revision with clearer arguments